vnext/Shared/WinRTWebSocketResource.cpp

```diff
@@ -145,67 +142,68 @@ IAsyncAction WinRTWebSocketResource::PerformConnect() noexcept
   self->m_connectRequested = false;
 }
 
-fire_and_forget WinRTWebSocketResource::PerformPing() noexcept
-{
+fire_and_forget WinRTWebSocketResource::PerformPing() noexcept {
   auto self = shared_from_this();
-  try
-  {
-    co_await resume_background();
 
-    co_await resume_on_signal(self->m_connectPerformed.get());
+  co_await resume_background();
 
-    if (self->m_readyState != ReadyState::Open)
-    {
-      self = nullptr;
-      co_return;
-    }
+  co_await resume_on_signal(self->m_connectPerformed.get());
 
+  if (self->m_readyState != ReadyState::Open) {
+    self = nullptr;
+    co_return;
+  }
+
+  try {
     self->m_socket.Control().MessageType(SocketMessageType::Utf8);
 
     string s{};
     winrt::array_view<const uint8_t> arr(
-      CheckedReinterpretCast<const uint8_t*>(s.c_str()),
-      CheckedReinterpretCast<const uint8_t*>(s.c_str()) + s.length());
+        CheckedReinterpretCast<const uint8_t *>(s.c_str()),
+        CheckedReinterpretCast<const uint8_t *>(s.c_str()) + s.length());
     self->m_writer.WriteBytes(arr);
 
-    co_await self->m_writer.StoreAsync();
+    auto async = self->m_writer.StoreAsync();
 
-    if (self->m_pingHandler)
-    {
-      self->m_pingHandler();
+    co_await lessthrow_await_adapter<DataWriterStoreOperation>{async};
+
+    auto result = async.ErrorCode();
+    if (result >= 0) {
```

Comment:
This is an HRESULT, right? Can we use `SUCCEEDED` macro or compare to `S_OK` to make intent clearer?

Reply:
Same comment in other places

Reply:
@asklar pointed out non-negative values could still indicate a non-success condition (i.e. `S_FALSE`). Hence, switched `SUCCEEDED()` out of this change.

Regarding using `0` instead of `S_OK`, @vmoroz mentioned this is the convention followed by C++/WinRT.
For example, see `<winrt/base.h>`, constructor `hresult_error(hresult const code, take_ownership_from_abi_t) noexcept`.